guides/octane-auth-torii-with-github.md

```diff
@@ -0,0 +1,668 @@
+later need to control actions:
+
+```sh
+ember generate route application
+ember generate route index
+ember generate route login
+ember generate controller application
+ember generate controller login
+```
+
+Ember-cli will ask about overwriting `app/templates/application.hbs`. That’s
+fine to do so. 
+
+Now let’s create our skeletal templates by editing the three files in
+`app/templates`:
+
+```handlebars
+{{!-- app/templates/application.hbs --}}
+<h1>My Ember Octane App</h1>
+<h2>Secret API Key: {{this.config.apiKey}}</h2>
+
+<p>
+  {{#if this.session.isAuthenticated}}
+    Authenticated <br>
+    <button onclick={{this.logout}}>Log Out</button>
+  {{else}}
+    Unauthenticated
+  {{/if}}
+</p>
+<hr>
+
+{{outlet}}
+```
+
+```handlebars
+{{!-- app/templates/index.hbs --}}
+<h3>Index Route</h3>
+{{#if this.session.data.authenticated.authorizationCode}}
+  <p>Authorization Code: {{this.session.data.authenticated.authorizationCode}}</p>
+{{/if}}
+```
+
+```handlebars
+{{!-- app/templates/login.hbs --}}
+<h3>Login Route</h3>
+<button>Log in to GitHub</button>
+```
+
+If you are familiar with Ember but unfamiliar with prefixing properties with
+`this.` in Handlebars templates, this is a new Ember convention reflected in
+Octane, where any property defined on the controller (or in a component) is
+referred to with `this.` in the template. `this.config` and `this.session` are
+properties that will be defined on the route controllers in later steps.
+
+When you spin up the server again with `ember serve`, you should see, at
+`http://localhost:4200`, something like this:
+
+![Initial Ember Octane App page](./assets/initial-ember-octane-app-page.png)
+
+Incidentally, while the server was spinning up, it probably complained that
+Torii has not yet been configured, so let’s do that.
+
+### Configure Torii
+
+Torii is configured in the file `config/environment.js`. Somewhere around the
+seventh line, perhaps underneath `rootURL: "/"`, add:
+
+```javascript
+    torii: {
+      sessionServiceName: "session",
+      providers: {
+        "github-oauth2": {
+          scope: "repo user",
+          apiKey: process.env.GITHUB_DEV_CLIENT_ID,
+          redirectUri: process.env.GITHUB_DEV_REDIRECT_URI,
+          tokenExchangeUri: process.env.DEV_TOKEN_EXCHANGE_URL
+        }
+      }
+    },
+```
+
+When you spin up the server again, it will not complain about Torii’s being
+unconfigured.
+
+## Implement the first stage of GitHub authorization
+
+OAuth is officially an authorization protocol, but is commonly used also for
+authentication when the initial authorization code is obtained over `https`.
+GitHub uses the OAuth [authorization code grant
+type](https://tools.ietf.org/html/rfc6749#section-4.1), which requires two
+steps. As noted at the top, the first step uses your client ID to get a
+temporary authorization code. The temporary authorization code acts as a
+single use bridge to authorization. 
+
+### Add `ember-simple-auth` Mixins to Routes
```

Comment:
Any chance you could add how to use a custom mixin ... I'm trying to override `authenticationRoute`